service/history/transferQueueTaskExecutorBase_test.go

```diff
@@ -0,0 +1,216 @@
+	"go.temporal.io/server/common/definition"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/common/namespace"
+	"go.temporal.io/server/common/primitives/timestamp"
+	"go.temporal.io/server/service/history/configs"
+	"go.temporal.io/server/service/history/consts"
+	"go.temporal.io/server/service/history/shard"
+	"go.temporal.io/server/service/history/tasks"
+	"go.temporal.io/server/service/history/tests"
+	"go.temporal.io/server/service/history/workflow"
+)
+
+func TestOnlyDeleteOpenExecutions(t *testing.T) {
+	testCases := []struct {
+		Name                    string
+		EnsureCloseBeforeDelete bool
+		CloseTransferTaskIdSet  bool
+		InActiveCluster         bool
+		MultiCursorQueue        bool
+		CloseTaskIsAcked        bool
+		ShouldDelete            bool
+	}{
+		{
+			Name:                    "skip the check",
+			EnsureCloseBeforeDelete: false,
+			ShouldDelete:            true,
+		},
+		{
+			Name:                    "no task id",
+			EnsureCloseBeforeDelete: true,
+			CloseTransferTaskIdSet:  false,
+			ShouldDelete:            true,
+		},
+		{
+			Name:                    "passive cluster",
+			EnsureCloseBeforeDelete: true,
+			CloseTransferTaskIdSet:  true,
+			InActiveCluster:         false,
+			ShouldDelete:            true,
+		},
+		{
+			Name:                    "single cursor queue unacked",
+			EnsureCloseBeforeDelete: true,
+			CloseTransferTaskIdSet:  true,
+			InActiveCluster:         true,
+			MultiCursorQueue:        false,
+			CloseTaskIsAcked:        false,
+			ShouldDelete:            false,
+		},
+		{
+			Name:                    "single cursor queue acked",
+			EnsureCloseBeforeDelete: true,
+			CloseTransferTaskIdSet:  true,
+			InActiveCluster:         true,
+			MultiCursorQueue:        false,
+			CloseTaskIsAcked:        true,
+			ShouldDelete:            true,
+		},
+		{
+			Name:                    "multicursor queue unacked",
+			EnsureCloseBeforeDelete: true,
+			CloseTransferTaskIdSet:  true,
+			InActiveCluster:         true,
+			MultiCursorQueue:        true,
+			CloseTaskIsAcked:        false,
+			ShouldDelete:            false,
+		},
+		{
+			Name:                    "multicursor queue acked",
+			EnsureCloseBeforeDelete: true,
+			CloseTransferTaskIdSet:  true,
+			InActiveCluster:         true,
+			MultiCursorQueue:        true,
+			CloseTaskIsAcked:        true,
+			ShouldDelete:            true,
+		},
+	}
+	for _, c := range testCases {
+		t.Run(c.Name, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockMutableState := workflow.NewMockMutableState(ctrl)
+			closeTransferTaskId := int64(10)
+			workflowKey := definition.NewWorkflowKey(uuid.New(), uuid.New(), uuid.New())
+			mockMutableState.EXPECT().GetExecutionInfo().Return(&persistencepb.WorkflowExecutionInfo{
+				NamespaceId:         workflowKey.NamespaceID,
+				WorkflowId:          workflowKey.WorkflowID,
+				CloseTransferTaskId: closeTransferTaskId,
+			}).AnyTimes()
```

Comment:
Any chances to specify times instead of all these `AnyTimes()` calls?

Reply:
Sure, I added it to all the calls which accept a ctx, since I assume those are the only ones which could take some time.